### PR TITLE
Feature/local play

### PR DESCRIPTION
**Pull request by gcandrade10**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- record videos
- create clones
- take over the world

#### Before/After Screenshots/Screen Record

- Before:
- After:

#### Fixes the following issue(s)

- Fixes #

#### Relies on the following changes

- 

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [ ] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -157,19 +157,19 @@ task runKtlint(type: JavaExec) {
...
-    classpath = configurations.ktlint
-    args "-F", "src/**/*.kt"
-}
+//task formatKtlint(type: JavaExec) {
+//    inputs.files(inputFiles)
+//    outputs.dir(outputDir)
+//    getMainClass().set("com.pinterest.ktlint.Main")
+//    classpath = configurations.ktlint
+//    args "-F", "src/**/*.kt"
+//}
```
Comment (mhmdanas) [inline app/build.gradle:166]: Same here.

File: app/build.gradle [hunk 2]
```diff
@@ -157,19 +157,19 @@ task runKtlint(type: JavaExec) {
...
 
 afterEvaluate {
     if (!System.properties.containsKey('skipFormatKtlint')) {
-        preDebugBuild.dependsOn formatKtlint
+//        preDebugBuild.dependsOn formatKtlint
     }
-    preDebugBuild.dependsOn runCheckstyle, runKtlint
+//    preDebugBuild.dependsOn runCheckstyle, runKtlint
```
Comment (mhmdanas) [inline app/build.gradle:172]: Ditto.

File: app/src/main/java/us/shandian/giga/service/DownloadManager.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ enum NetworkState {Unavailable, Operating, MeteredOperating}
      * @param context Context for the data source for finished downloads
      * @param handler Thread required for Messaging
      */
-    DownloadManager(@NonNull Context context, Handler handler, StoredDirectoryHelper storageVideo, StoredDirectoryHelper storageAudio) {
+    public DownloadManager(@NonNull Context context, Handler handler, StoredDirectoryHelper storageVideo, StoredDirectoryHelper storageAudio) {
```
Comment (mhmdanas) [inline app/src/main/java/us/shandian/giga/service/DownloadManager.java:65]: Apologies if it's obvious, but why was this made public?

File: foo.patch [hunk 4]
```diff
@@ -0,0 +1,2006 @@
+From 6d1eee441f3051c730cc4b51294bc83fcc1d156e Mon Sep 17 00:00:00 2001
```
Comment (mhmdanas) [inline foo.patch:1]: Please remove this file.

File: gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,6 +1,6 @@
+#Fri Jun 03 09:13:20 COT 2022
 distributionBase=GRADLE_USER_HOME
-distributionPath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
-distributionSha256Sum=e6d864e3b5bc05cc62041842b306383fc1fefcec359e70cebb1d470a6094ca82
-zipStoreBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
```
Comment (mhmdanas) [inline gradle/wrapper/gradle-wrapper.properties:6]: What's the purpose behind these changes?

**Review by mhmdanas — CHANGES_REQUESTED** [review]
Can you clarify the PR title a bit? I believe this PR implements local file playback, but the title isn't exactly clear on that. Also, please fill out the PR template.

**Review by TobiGr — CHANGES_REQUESTED** [review]
Thank you for PR. please fill in the template and add comments to your code explaining the general purpose of methods and if needed, what they do specifically. This makes the review process easier for us and keeps the code somewhat maintainable in the future.

**Comment by opusforlife2** [comment]
Didn't we reject local file playback as 'out of scope'?

**Comment by opusforlife2** [comment]
Since there is a contributor willing to maintain this feature, I think a 'wait and see' approach could be taken. What is important is for the feature to work _without_ breaking, within whatever scope is decided to be adopted for it.

For example, the feature could be highly limited to only playing back a single file from the SAF file browser, like file browsing apps themselves often do.

Or it could create a folder within the app directory (as a narrow option) or somewhere else on the filesystem (as a broader option) and manage and play files within that directory/folder.

And then there are even bigger scopes, of course. It could ask the user to mark folders to watch for media, and keep track of file addition/removal/modification.

@gcandrade10 I have no idea what your PR does right now. Please fill out the template properly.

**Comment by litetex** [comment]
No feedback from the author since a month - seems to be a fire and forget PR.

And btw this PR is violating our statement from https://github.com/TeamNewPipe/NewPipe/issues/478.

Closing.

**Comment by Sepero** [comment]
It's reality unfortunate to see submissions like this be dismissed. NewPipe is a great YouTube frontend, but it's also a great Video Player in general. It would be great if NewPipe could play local videos (even if only supports a few formats like mp4 and webm). I actually much prefer NewPipe's interface over VLC

**Comment by opusforlife2** [comment]
@Sepero https://newpipe.net/FAQ/#bugs-play-downloaded-videos

**Pull request closed without merging**